axon-server-connector/src/main/java/org/axonframework/axonserver/connector/ErrorCode.java

```diff
@@ -116,18 +134,24 @@ public AxonException convert(ErrorMessage errorMessage){
      * @param throwable the descriptor of the error
      * @return the Axon Framework exception
      */
-    public AxonException convert(Throwable throwable){
+    public AxonException convert(Throwable throwable) {
         return convert("", throwable);
     }
 
     /**
      * Converts the {@code source} and the {@code throwable} to the relevant AxonException
      *
-     * @param source The location that originally reported the error
+     * @param source    The location that originally reported the error
      * @param throwable the descriptor of the error
      * @return the Axon Framework exception
      */
-    public AxonException convert(String source, Throwable throwable){
-        return convert(ExceptionSerializer.serialize(source, throwable));
+    public AxonException convert(String source, Throwable throwable) {
+        return convert(ExceptionSerializer.serialize(source, throwable), () -> HandlerExecutionException.resolveDetails(throwable).orElse(null));
+    }
+
+    private interface ExceptionBuilder {
```

Comment:
I'd personally prefer a little javadoc here, for other developers and such. Not overly important by the way, just a tad bit cleaner.